notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+$$
+
+Our goal is to find the $\hat{\theta}$ that results in the line with the least squared error $L(\hat{\theta}, \vec{x}, \vec{y})$:
+
+$$
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&= \sum_{i = 1}^{n}(y_i - f_\hat{\theta} (x_i))^2\\
+&= \sum_{i = 1}^{n}(y_i - \hat{\theta_0} - \hat{\theta_1} x_i)^2\\
+&= \sum_{i = 1}^{n}(y_i - \begin{bmatrix} 1 & x_i \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix} ) ^2
+\end{aligned}
+$$
+
+In our data set, n = 3, so let's expand out this equation by breaking up the summation.
+
+$$
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&=
+(y_1 - \begin{bmatrix} 1 & x_1 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2  \\
+&+
+(y_2 - \begin{bmatrix} 1 & x_2 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2 \\
+&+
+(y_3 - \begin{bmatrix} 1 & x_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2 \\
+\end{aligned}
+$$
+
+Notice that our loss function is a sum of squares. Now, notice that the *L2*-norm for a vector has a similar form: 
+
+$$\Vert \vec{v} \Vert = \sqrt{v_1^2 + v_2^2 + \dots + v_n^2}$$ 
+
+So if we can express each $y_i - \begin{bmatrix} 1 & x_i \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}$ as a vector, we can express our loss as the *L2*-norm, squared:
+
+$$ 
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & x_1 \\ 1 & x_2 \\ 1 & x_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+X 
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+\vec{\hat{y}}
+\qquad \right \Vert ^2 \\
+\end{aligned}
+$$
+
+This gives us a new perspective on what it means to minimize the least squares error.
+
+
+$X$ and $\vec{y}$ are fixed, but $\hat{\theta_0}$ and $\hat{\theta_1}$ can take on any value, so there are *infinite possible vectors* created by matrix multiplying $X$ and $\hat{\theta}$. To have the smallest loss, we want to choose $\hat{\theta_0}$ and $\hat{\theta_1}$ such that the resulting vector is as close to $\vec{y}$ as possible. This is the same underlying logic as before: find the smallest squared distance; but between two vectors $\vec{y}$ and $\vec{\hat{y}}$, rather than considering the indefinite number of data points. Lastly, notice that the matrix multiplication is a linear combination of the columns of $X$: each $\hat{\theta_i}$ only ever multiplies with one column of $X$--this perspective shows us that our $\hat{y}$ is a linear combination of the features of our data. Let's get some geometric intuition on these observations.
```

Comment:
Hmm.. I don't really like the infinite possible vectors part. I think it'd be better to say that the vector y hat can take on any value that is a linear combination of the columns of X?

Also, "this is the same underlying logic as before:". What is "before" referring to? I don't remember

Reply:
Hmm.. I don't really know how to fix it, but the last sentence sounds a bit too casual. Idk if Sam would like it